File: 프로그래머스_소수찾기.py
import math
def solution(n):
    primeNum=[True for i in range(n+1)]
    i=2
    primeNum[0]=False
    primeNum[1]=False
    for i in range(2,int(math.sqrt(n))+1):
        if primeNum[i]:
            for j in range(2*i,n+1,i):
                primeNum[j]=False
    prime=[i for i in primeNum if i==True]
    answer = len(prime)
    return answer

File: programmers/프로그래머스_소수찾기.cpp
#include <string>
#include <vector>

using namespace std;
int PrimeNum_initer(int n){
    vector <int> Prime;
    bool PrimeNum[1000001]={false}; 
    for(int i=2;i<=n;i++)
        PrimeNum[i]=true;

    for(int i=2;i*i<=n;i++){
        if(PrimeNum[i]){
         for(int j=2;j*i<=n;j++)
            PrimeNum[i*j]=false;
        }
    }
    for(int i=2;i<=n;i++){
        if(PrimeNum[i])
            Prime.push_back(i);
    }
    return Prime.size();
}
int solution(int n) {
    
    int answer = PrimeNum_initer(n);
    return answer;
}
